gobblin-data-management/src/main/java/gobblin/data/management/version/FileStatusTimestampedDatasetVersion.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.data.management.version;
+
+import java.util.Set;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.joda.time.DateTime;
+
+import com.google.common.collect.Sets;
+
+/**
+ * A {@link TimestampedDatasetVersion} that is also aware of the {@link FileStatus}s of all its paths.
+ */
+public class FileStatusTimestampedDatasetVersion extends TimestampedDatasetVersion implements FileStatusAware {
```

Comment:
It will be a cascading change and I want to limit the change to this `VersionFinder`. Making `TimestampedDatasetVersion` implement `FileStatusAware` would mean all the `VersionFinder`s that build a `TimestampedDatasetVersion` needs to be changed.